eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -1332,6 +1332,13 @@
     double injectInto(double injectedValue, DoubleObjectToDoubleFunction<? super T> function);
 
     /**
+     * Adds all the elements in this iterable to the specific target Collection.
+     *
+     * @since 8.0
+     */
+    <R extends Collection<T>> R into(R target);
```

Comment:
I think a more flexible signature would be:

```java
<R extends Collection<? super T>> R into(R target);
```

This would allow collecting into a `MutableCollection<Object>` for example.